src/core/components/ChatHistory.tsx

```diff
@@ -158,11 +158,23 @@ const HistoryItem = memo(
       const trimmedTitle = newTitle.trim();
       if (trimmedTitle === '') {
         setNewTitle(title);
-      } else {
-        saveToLocalStorage(trimmedTitle);
-        setNewTitle(trimmedTitle);
+        return;
       }
+
+      //  Update UI first
+      setNewTitle(trimmedTitle);
       setEditingTitle(false);
```

Comment:
I made a slight UX change here. Once the user saves the new title, we exit them out of editing mode